### PR TITLE
Update gorm version and fix wildcards on non strings

### PR DESCRIPTION
**Pull request by bramca**
- Updated the gorm version: 
the way clauses are set changed a lot in the new version, so there were some code changes needed to keep it working like it used to
- fix wildcards on non strings: 
some DBs only accept LIKE queries on varchars. A cast to varchar was added so this plugin does not break on non string fields e.g. uuids, datetimes...

### CODE REVIEW

File: query.go [hunk 1]
```diff
@@ -118,22 +107,46 @@ func (d *gormLike) queryCallback(db *gorm.DB) {
...
+				query = query.Or(condition, value)
 			}
 
 			// Don't alter the query if it isn't necessary
 			if likeCounter == 0 {
 				continue
 			}
 
-			exp.Exprs[index] = db.Session(&gorm.Session{NewDB: true}).Where(query).Statement.Clauses["WHERE"].Expression
+			// This feels a bit like a dirty hack
```
Comment (survivorbat) [inline query.go:118]: Nice documentation!

File: query.go [hunk 2]
```diff
@@ -118,22 +107,46 @@ func (d *gormLike) queryCallback(db *gorm.DB) {
...
+			newExpression.Exprs = query.Statement.Clauses["WHERE"].Expression.(clause.Where).Exprs
+
+			expressions[index] = newExpression
 		}
 	}
+
+	return expressions
+}
+
+//nolint:gocognit,cyclop // Acceptable
```
Comment (survivorbat) [inline query.go]: No longer necessary, it's smaller now

File: query_test.go [hunk 3]
```diff
@@ -35,40 +35,40 @@ func TestGormLike_Initialize_TriggersLikingCorrectly(t *testing.T) {
 			query:    defaultQuery,
 		},
 
-		// Check if everything still works
+		// // Check if everything still works
```
Comment (survivorbat) [inline query_test.go]: double comment

File: query_test.go [hunk 4]
```diff
@@ -141,27 +149,26 @@ func TestGormLike_Initialize_TriggersLikingCorrectly(t *testing.T) {
...
-			existing: []ObjectA{{Name: "jessica", Age: 53}, {Name: "amy", Age: 20}, {Name: "John", Age: 25}},
-			expected: []ObjectA{{Name: "jessica", Age: 53}, {Name: "amy", Age: 20}},
+			existing: []ObjectA{{ID: uuid.MustParse("e8a2e524-dcb5-4436-b608-46f38ab47944"), Name: "jessica", Age: 53}, {ID: uuid.MustParse("a3367b2f-0d13-4e64-afa2-7d50844e370d"), Name: "amy", Age: 20}, {ID: uuid.MustParse("fc5e4af9-5959-4e31-8b56-d33fdb6d71c3"), Name: "John", Age: 25}},
+			expected: []ObjectA{{ID: uuid.MustParse("e8a2e524-dcb5-4436-b608-46f38ab47944"), Name: "jessica", Age: 53}, {ID: uuid.MustParse("a3367b2f-0d13-4e64-afa2-7d50844e370d"), Name: "amy", Age: 20}},
 			options:  []Option{WithCharacter("🍌")},
 		},
 		"more complex like query with 🍓": {
 			filter: map[string]any{
-				"name": "🍓a🍓",
+				"name": "%a%",
```
Comment (survivorbat) [inline query_test.go]: This change seems incorrect

File: query_test.go [hunk 5]
```diff
@@ -77,51 +77,59 @@ func TestGormLike_Initialize_TriggersLikingCorrectly(t *testing.T) {
...
 		},
 		"more complex multi-value, all like queries": {
 			filter: map[string]any{
 				"name":  []string{"%a%", "%o%"},
 				"other": []string{"%ooo", "aaa%"},
 			},
 			query:    defaultQuery,
-			existing: []ObjectA{{Name: "jessica", Age: 53, Other: "aaaooo"}, {Name: "amy", Age: 20, Other: "aaaooo"}, {Name: "John", Age: 25, Other: "aaaooo"}},
-			expected: []ObjectA{{Name: "jessica", Age: 53, Other: "aaaooo"}, {Name: "amy", Age: 20, Other: "aaaooo"}, {Name: "John", Age: 25, Other: "aaaooo"}},
+			existing: []ObjectA{{ID: uuid.MustParse("77c5e4f2-30a9-459b-8178-0a3b61369cdc"), Name: "jessica", Age: 53, Other: "aaaooo"}, {ID: uuid.MustParse("909ec358-ee69-4f74-8e70-069311125e8a"), Name: "amy", Age: 20, Other: "aaaooo"}, {ID: uuid.MustParse("37ac3821-c4e7-4004-b2ed-2cdc7c9ffb53"), Name: "John", Age: 25, Other: "aaaooo"}},
```
Comment (survivorbat) [inline query_test.go]: Could you multiline this data? I don't think the uuid makes it more readable

File: query_test.go [hunk 6]
```diff
@@ -77,51 +77,59 @@ func TestGormLike_Initialize_TriggersLikingCorrectly(t *testing.T) {
...
+				"id": "%22%",
+			},
+			query:    defaultQuery,
+			existing: []ObjectA{{ID: uuid.MustParse("573fc37d-622c-4d41-b48b-201bf195a1c9"), Name: "jessica", Age: 53}, {ID: uuid.MustParse("66df272f-5667-4472-950a-2269833fdbab"), Name: "amy", Age: 20}, {ID: uuid.MustParse("3e3b8dbc-37a3-4113-b05c-6c1d5a3ad5fa"), Name: "John", Age: 25}},
+			expected: []ObjectA{{ID: uuid.MustParse("573fc37d-622c-4d41-b48b-201bf195a1c9"), Name: "jessica", Age: 53}, {ID: uuid.MustParse("66df272f-5667-4472-950a-2269833fdbab"), Name: "amy", Age: 20}},
 		},
 		"more complex like query": {
 			filter: map[string]any{
-				"name": "%a%",
+				"name": "a%",
```
Comment (survivorbat) [inline query_test.go]: why is this being replaced?

File: query_test.go (hunk 5 above)
Comment (survivorbat) [reply query_test.go]: Or perhaps the IDs could be removed as to not pollute the test data where it isn't needed

File: query_test.go [hunk 7]
```diff
@@ -279,7 +286,8 @@ func TestGormLike_Initialize_TriggersLikingCorrectly(t *testing.T) {
 			assert.NoError(t, err)
 
 			var actual []ObjectA
-			err = testData.query(db).Where(testData.filter).Find(&actual).Error
+			dbQuery := testData.query(db).Where(testData.filter).Find(&actual)
```
Comment (survivorbat) [inline query_test.go]: why this change? Leftover? ;)

**Review by survivorbat — COMMENTED** [review]
Small nitpick comments

File: query_test.go (hunk 7 above)
Comment (bramca) [reply query_test.go]: indeed, removed

File: query_test.go (hunk 6 above)
Comment (bramca) [reply query_test.go]: probably a mistake, reverted

File: query_test.go (hunk 5 above)
Comment (bramca) [reply query_test.go]: made the test inputs more readable

File: query_test.go (hunk 4 above)
Comment (bramca) [reply query_test.go]: indeed, reverted

File: query_test.go (hunk 3 above)
Comment (bramca) [reply query_test.go]: removed

File: query.go (hunk 2 above)
Comment (bramca) [reply query.go]: removed

File: query.go (hunk 1 above)
Comment (bramca) [reply query.go:118]: thanks!
